src/uhs/atomizer/sentinel/controller.cpp

```diff
@@ -83,30 +109,81 @@ namespace cbdc::sentinel {
             send_transaction(tx);
         }
 
-        return response{status, res};
+        return execute_response{status, res};
     }
 
     void controller::send_transaction(const transaction::full_tx& tx) {
-        const auto compact_tx = cbdc::transaction::compact_tx(tx);
-        auto ctx_pkt = std::make_shared<cbdc::buffer>();
-        auto ctx_ser = cbdc::buffer_serializer(*ctx_pkt);
-        ctx_ser << compact_tx;
+        auto compact_tx = cbdc::transaction::compact_tx(tx);
+        auto attestation = compact_tx.sign(m_secp.get(), m_privkey);
+        compact_tx.m_attestations.insert(attestation);
+
+        gather_attestations(tx, std::move(compact_tx), {});
```

Comment:
cc @metalicjames

Reply:
I moved the send logic into its own method for both the atomizer and 2PC sentinels.

Reply:
This update still leaves the actual call to send the inputs inside `gather_attestations()` (which is the part that I find confusing)—on the surface, I wouldn't expect gathering attestations to require sending the inputs after the attestations have been gathered. Those feel like discrete steps to me (such that I'd expect the `send_compact_tx()` call to take place at the end of `send_transaction()` *after* `gather_attestations()` rather than those two concepts being coupled together).

Does that make sense? Is there an aspect to why gathering the attestations necessitates forwarding the inputs immediately that I'm missing?

Reply:
Where else would you put it? `gather_attestations` is a series of callbacks to request the attestations from remote sentinels, which will respond asynchronously. Once we receive the last callback, we have enough attestations and are ready to send the transaction. `gather_attestations` (and also `send_transaction`) returns right away after sending the first attestation request, it does not block and return the attestation.

Reply:
That's quite clarifying; sorry for the confusion there. I suppose then it's less about anything than the name (it's not so much gathering attestations as it is actually sending the transaction, the first step of which is to gather the attestations). I'll call it resolved for now. Thank you!